frontend/www/js/omegaup/problem/details.ts

```diff
@@ -90,23 +90,19 @@ OmegaUp.on('ready', () => {
           hasBeenNominated: this.hasBeenNominated,
         },
         on: {
-          'apply-filter': (filter: string, value: string) => {
-            const filterSelected: RunFilters = {};
-            switch (filter) {
-              case 'verdict':
-                filterSelected.verdict = value;
-                break;
-              case 'language':
-                filterSelected.language = value;
-                break;
-              case 'username':
-                filterSelected.username = value;
-                break;
-              case 'status':
-                filterSelected.status = value;
-                break;
+          'apply-filter': (
+            filter: 'verdict' | 'language' | 'username' | 'status',
+            value: string,
+          ) => {
+            if (!value) {
+              runsStore.commit('removeFilter', filter);
+              refreshRuns();
+              return;
             }
-            runsStore.commit('applyFilter', filterSelected);
+            runsStore.commit('applyFilter', {
+              [filter]: value,
+            } as RunFilters);
+            refreshRuns();
```

Comment:
```suggestion
            if (value) {
              runsStore.commit('applyFilter', {
                [filter]: value,
              } as RunFilters);
            } else {
              runsStore.commit('removeFilter', filter);
            }
            refreshRuns();
```

aquí ambas ramas tienen "el mismo peso", entonces el guard clause pattern no es obligatorio.

Reply:
Listo.